drgn/helpers/common/stack.py

```diff
@@ -0,0 +1,76 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+"""
+Stack
+-----
+
+The ``drgn.helpers.common.stack`` module provides helpers for working with the
+stack trace.
+"""
+
+from drgn import PlatformFlags, StackTrace
+from drgn.helpers.common.memory import identify_address
+
+__all__ = ("print_annotated_stack",)
+
+
+def print_annotated_stack(trace: StackTrace) -> None:
+    """
+    Print the content of the stack trace, identifying symbols and
+    slab objects.
+
+    :param trace: Stack trace to print.
+    """
+    prog = trace.prog
+    # platform must be known if there is a stack trace
+    assert prog.platform is not None
+
+    word_size = 8 if prog.platform.flags & PlatformFlags.IS_64_BIT else 4
+
+    frames = [frame for frame in trace if not frame.interrupted]
+    frames_addrs = [frame.register("rsp") for frame in frames]
+    hex_format = "{:016x}" if word_size == 8 else "{:08x}"
+
+    start_addr = frames_addrs[0]
+    end_addr = frames_addrs[-1] + word_size - 1
+    stack_size = end_addr - start_addr + 1
+
+    stack_bytes = prog.read(start_addr, stack_size)
+    frame_ind = 0
+
+    for offset in range(0, len(stack_bytes), word_size):
+        addr = start_addr + offset
+        hex_addr = hex_format.format(addr)
+
+        word_bytes = stack_bytes[offset : offset + word_size]
+
+        if prog.platform.flags & PlatformFlags.IS_LITTLE_ENDIAN:
+            word_val = int.from_bytes(word_bytes, "little")
+        else:
+            word_val = int.from_bytes(word_bytes, "big")
+
+        # multiple frames can have the same address
+        # for e.g, the first context_switch frame
```

Comment:
FYI, these stack frames with the same address are typically because of inlined functions.

Reply:
Makes sense!